content/sensu-core/1.0/guides/pre-compile-plugins.md

```diff
@@ -0,0 +1,146 @@
+We'll be using `sensu-plugins-aws` as it requires `nokogiri`,  usually requiring a longer build time.
+
+# Prerequist
+
+Sources:
+* [GitHub - luislavena/gem-compiler: A RubyGems plugin that generates binary gems](https://github.com/luislavena/gem-compiler)
+This guide assumes the following:
+* Centos 7 platform
+* You have some way to store these artifacts and make them available for systems in your environment
+_NOTE: Compiled gems are tied to the Ruby version they are compiled for. If Sensu upgrade Ruby 2.4 -> Ruby 2.5, packages will need to be rebuilt._
+_NOTE: Compiled gems are tied to CPU architecture for which they are compiled on._
+
+## Installing The Build Toolchain
+
+We'll first add the Sensu yum repository definition and install Sensu
+
+```bash
+printf '[sensu]
+name=sensu
+baseurl=https://repositories.sensuapp.org/yum/$releasever/$basearch/
+gpgcheck=0
+enabled=1' | sudo tee /etc/yum.repos.d/sensu.repo && yum install -y sensu
+```
+
+And then install EPEL and Dev tools
+
+```bash
+sudo yum install -y epel-release
+sudo yum groupinstall -y "Development Tools"
+```
+
+Finally we'll add Sensu's embedded bin to our path
+
+```bash
+export PATH=$PATH:/opt/sensu/embedded/bin
+```
+
+## Installing a Gem to a tmp directory
+
+Install the gem-compiler package
+
+```bash
+gem install gem-compiler
+```
+
+We'll now install the source gems into a cache directory
+
+```bash
+mkdir /tmp/gems
+gem install --no-ri --no-rdoc --install-dir /tmp/gems sensu-plugins-aws
+```
+
+Observing the `gem install` output, we can see that nokogiri and unf_ext build native extensions.
+These are the gems we will compile so we can install them from our own repository.
+
+### Compile the Gems for packaging
+
+We'll use `gem-compiler` to rebuild the gem packages with the included native extensions:
+
+```bash
+cd /tmp/gems/cache
+gem compile unf_ext-0.0.7.4.gem
+gem compile nokogiri-1.8.1.gem --prune
+```
+
+_NOTE: `nokogiri` requires --prune flag per gem-compiler readme notes _
+
+The resulting compiled gem files have a x86_64-linux in their file name denoting their system architech:
+
+```bash
+$ ls  -1 *-x86_64-linux.gem
+nokogiri-1.8.1-x86_64-linux.gem
+unf_ext-0.0.7.4-x86_64-linux.gem
+```
+
+## Install Pre-Compiled Gems on Targeted System and Testing
+
+### Install compiled gem into embedded Ruby
+Now we can copy these gems to another system, install them into the sensu embedded ruby. 
+In a production or development enviroment, you'd want to put these in a rubygems server of your own (e.g. [Geminabox(https://github.com/geminabox/geminabox), [Artifactory](https://jfrog.com/artifactory/) or similar). You can also use a tool like `fpm` to convert the gems into a system package.
+
+Here we've copied the compiled gems to a system with Sensu installed, but no gcc or other compile toolchain
+
+ _NOTE: nokogiri needs libxml2 and possibly libxslt to be present on your system._
```

Comment:
I feel like we shouldn't leave a "possibly" in here. We should know without a doubt as to what the dependencies are and what needs to be present. The "possibly" here makes it come across that we don't know what we're talking about.

Reply:
removed "possibly"

Reply:
added it to the install process